src/my_react/useState.ts

```diff
@@ -0,0 +1,44 @@
+import App from 'src/App';
+import render from './render';
+
+/* DOM 전체를 리렌더링, diffing은 다음에 구현해보기 */
+const rerender = () => {
+  const $app = document.getElementById('app');
+  if ($app instanceof HTMLElement) {
+    $app.innerHTML = '';
+    render($app, App()); // 이후 Diffing 알고리즘 구현하는 로직 추가할 땐 App을 import해와서 인자로 전달하는 게 아니라, vDOM을 diffing해서 변경 필요한 곳만 전달하는 게 필요
+  }
+};
+
+// 즉시 실행 함수를 반환해서 states와 cursor를 클로저로 관리
+const useState = (() => {
+  // useState의 실행 순서에 따라 정의된 상태들 배열로 관리
+  const states: unknown[] = [];
+  let cursor = 0;
+
+  return <T>(init: T | (() => T)) => {
```

Comment:
[P3] 값이 안들어오는 경우도 존재할거라 옵셔널로 두는 것이 react와 더 비슷할 것 같습니다!

```suggestion
  return <T>(init?: T | (() => T)) => {
```

Reply:
그렇네요 이것도 생각 못해봤던 부분인데ㅎㅎ 감사합니다!